plugins/database/src/db_plugin.cpp

```diff
@@ -15757,6 +15723,45 @@ auto db_update_ticket_write_byte_count_op(
     return SUCCESS();
 } // db_update_ticket_write_byte_count_op
 
+auto db_get_creator_of_delay_rule_op(
+    irods::plugin_context& _ctx,
+    const char*            _rule_id,
+    char**                 _username,
+    char**                 _zone) -> irods::error
+{
+    if (const auto ret = _ctx.valid(); !ret.ok()) {
+        return PASS(ret);
+    }
+
+    if (!_rule_id || !_username || !_zone) {
+        return ERROR(SYS_INVALID_INPUT_PARAM, "Invalid input: rule id, username, or zone is null");
+    }
+
+    auto [db_instance, db_conn] = irods::experimental::catalog::new_database_connection();
+
+    nanodbc::statement stmt{db_conn};
+    nanodbc::prepare(stmt, "select exe_context from R_RULE_EXEC where rule_exec_id = ?");
```

Comment:
Could this be done with a GenQuery? Or is there a reason we would want it in a database operation?

Reply:
as i understand it, genquery can't get the context - could be too big to serialize...

Reply:
That's correct. It's more so has to do with the db plugin's implementation. The db plugin isn't designed to handle columns holding large amounts of data.